### ISSUE TITLE
Attachments page: verification statement logic

### ISSUE THREAD

**Issue by pbastia** [bcgov/cas-reporting#453]
#### Describe the task

Right now the verification statement is required no matter what, but that logic will need to be amended to be the real one
Followup to #445 

#### Acceptance Criteria

Given I am an industrial reporter
And my reporting circumstances require that I submit a verification statement
When I am on the Attachments page
Then I can't submit the page until I have uploaded a file in the verification statement form.

Given I am an industrial reporter
And my reporting circumstances DOES NOT require that I submit a verification statement
When I am on the Attachments page
Then I can submit the page without an uploaded a file in the verification statement form.

#### Development Checklist:

- [x] In file `bciers/apps/reporting/src/app/components/attachments/AttachmentsPage.tsx` implement `getReportNeedsVerification()` using response value for `AttachmentForm` prop `isVerificationStatementMandatory` 
- [x] Update `AttachmentPage`  vitest

#### Additional context

- Add any other context about the task here.
- Or here

**Comment by rdromey** [comment]
@shon-button to write dev checklist + AC

**Comment by shon-button** [comment]
@patriciarussellCAS 

This should be tested with release v1.18.0 - after the refactor of conditions for when to display or not addressed in [PR 2625](https://github.com/bcgov/cas-registration/pull/2625)

[PR Test Steps](https://github.com/bcgov/cas-registration/pull/2625)
Replace `http://localhost:3000/` with `https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/`

**Comment by patriciarussellCAS** [comment]
I was able to confirm that I could not proceed past the attachments page when I have a verification requirement so that works. 
I was not able to test whether or not I would be blocked on the attachments page if I did NOT have a verification requirement. 
@fviduya did you happen to test this step as part of your review? (despite the bug that was found when clicking "back") If you were a reporting operation or potential reporting operation did you get a warning on the attachments page to upload a doc?

**Comment by fviduya** [comment]
Hi @patriciarussellCAS, I tested it under 450, 405, and 493 but wasn't able to document it. I tested it quickly, [here's](https://bcgov-my.sharepoint.com/:w:/g/personal/francesca_viduya_gov_bc_ca/EairSX5iRo1In80epuQM2CgB82w4rzRyiFYUAC387woO-A?e=mFG3IH) the test document for your reference. 

I can confirm that with Potential Reporting Operation and Reporting Operation (Emissions attributable for reporting < 25000 tc02e), I did not get any warning when I clicked Submit on the Sign-off page.

**Comment by patriciarussellCAS** [comment]
Thanks! I've noticed that the compliance summary page is being used to determine the conditions for verification, but this won't work for reporting operations, potential reporting operations or EIOs as they do NOT have a compliance summary in their workflow. Should this be a new ticket!!
